app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -50,6 +54,44 @@ class CrimeFragment : Fragment(), FragmentResultListener {
         super.onViewCreated(view, savedInstanceState)
         childFragmentManager.setFragmentResultListener(DIALOG_DATE, viewLifecycleOwner, this)
         observeData()
+        crime = Crime()
```

Comment:
This crime should be in the ViewModel.
We should first try to fetch it from database based on the crime ID, if it is not present, only then create a new instance.

Reply:
We can move the on Click listener implementations that update the crime model to the view model in order to have access to the crime model